### ISSUE TITLE
Setting a group as a group member breaks has_members associations on that group

### ISSUE THREAD

**Issue by juhazi** [dwbutler/groupify#45]
The use case I tried to implement is:

- have Teams, Users, and Tags as AR models
- Teams and Tags inherit from Groups, which are groupify groups
- Users are `groupify :group_member`
- Users are members of Teams and Tags
- Teams are also members of Tags

issue can be demonstrated by making the following change to active_record_spec.rb
```
class Group < ActiveRecord::Base
  groupify :group, members: [:users, :widgets, "namespaced/members"], default_members: :users
end

class Organization < Group
  groupify :group_member #TODO: breaks has_many :managers 

  has_members :managers
end
```

results are following failures:
```
  1) Groupify::ActiveRecord when using groups finds all members of group
     Failure/Error: expect(organization.managers).to include(manager)
     
       expected #<ActiveRecord::Associations::CollectionProxy []> to include #<Manager id: 1, name: nil, type: "Manager">
       Diff:
       @@ -1,2 +1,2 @@
       -[#<Manager id: 1, name: nil, type: "Manager">]
       +[]
       
     # ./spec/active_record_spec.rb:234:in `block (3 levels) in <top (required)>'

  2) Groupify::ActiveRecord when using groups when merging groups merges incompatible groups as long as all the source members can be moved to the destination
     Failure/Error: expect(destination.users.to_a).to include(user)
     
       expected [] to include #<User id: 1, name: nil, type: nil>
       Diff:
       @@ -1,2 +1,2 @@
       -[#<User id: 1, name: nil, type: nil>]
       +[]
       
     # ./spec/active_record_spec.rb:378:in `block (4 levels) in <top (required)>'

  3) Groupify::ActiveRecord when using groups when merging groups moves the members from source to destination and destroys the source
     Failure/Error: expect(destination.managers).to include(manager)
     
       expected #<ActiveRecord::Associations::CollectionProxy []> to include #<Manager id: 2, name: nil, type: "Manager">
       Diff:
       @@ -1,2 +1,2 @@
       -[#<Manager id: 2, name: nil, type: "Manager">]
       +[]
       
     # ./spec/active_record_spec.rb:341:in `block (4 levels) in <top (required)>'
```

The bug seems to be in this generated SQL query:
```
SELECT DISTINCT "users".*
FROM "users"
INNER JOIN "group_memberships"
ON "users"."id" = "group_memberships"."member_id"
WHERE "group_memberships"."member_id" = ?
AND "group_memberships"."member_type" = ?
AND "group_memberships"."member_type" = 'User'
[["member_id", 2], ["member_type", "Group"]]
```

As is apparent, it will never return any matching records.

**Comment by dwbutler** [comment]
Groupify is not (currently) designed to allow the same model to be both a group and a group member. I'll look into this more. The culprit seems to be that both `GroupMember` and `Group` define their own association with the same name - `group_memberships`.

**Comment by juhazi** [comment]
I split the association into two in the referenced commit above. This is of course not compatible with groupify 0.x versions and also differs in exposed associations in the mongoid adapter.

The tests seems to pass after trivial changes to direct `group_memberships` association usage.

Would you consider this a reasonable approach?

**Comment by dwbutler** [comment]
This is basically the approach I was going to try. Please go ahead and submit as a PR.

I'm fine with changing the association name because I've always recommended not using the `group_memberships` association directly. It's an internal implementation detail and not part of the public API. I'll make sure this is made crystal clear in the docs.

The Mongoid adapter, for example, implements this completely differently by embedding the memberships only on the group member in a `group_memberships` key. The group doesn't have a `group_memberships` relation at all. But the public API is exactly the same.

**Comment by dwbutler** [comment]
The fix has been released in v0.8.0. Thanks for your contribution!
